t/cli.t

```diff
@@ -58,71 +62,108 @@ subtest 'local-config' => sub {
     my $config = dist_file 'NewFangle-Agent', 'test.yml';
 
     subtest 'Default' => sub {
-        like test('local-config', '-'),
-            qr/  threshold: is_apdex_failing/, 'Prints default config as YAML';
+        like test( [ 'local-config', '-' ] ),
+            qr/  threshold: is_apdex_failing/,
+            'Prints default config as YAML';
     };
 
     subtest 'From file' => sub {
-        like test('local-config', $config),
+        like test( [ 'local-config', $config ] ),
             qr/^license_key: DEADBEEF/m, 'Prints config as YAML';
 
         local $ENV{NEWRELIC_LICENSE_KEY} = 'foo';
 
-        like test('local-config', $config),
+        like test( [ 'local-config', $config ] ),
             qr/^license_key: foo/m, 'Prints local config as YAML';
     };
 };
 
 subtest 'generate-config' => sub {
     subtest 'Print to output' => sub {
-        my $out = test('generate-config', 'foo');
+        my $out = test( [ 'generate-config', 'foo' ] );
 
-        like $out, qr/^# This file configures the unofficial New Relic Perl Agent/m,
+        like $out,
+            qr/^# This file configures the unofficial New Relic Perl Agent/m,
             'Generates a config file';
-        like $out, qr/^license_key: 'foo'/m, 'Prints config with new license key';
+        like $out, qr/^license_key: 'foo'/m,
+            'Prints config with new license key';
     };
 
     subtest 'Print to path' => sub {
         my $path = tempfile();
 
-        is test('generate-config', 'foo', $path), '', 'Nothing to STDOUT';
+        is test( [ 'generate-config', 'foo', $path ] ), '',
+            'Nothing to STDOUT';
 
         my $out = do {
             open my $fh, '<', $path or die;
             local $/, <$fh>;
         };
 
-        like $out, qr/^# This file configures the unofficial New Relic Perl Agent/m,
+        like $out,
+            qr/^# This file configures the unofficial New Relic Perl Agent/m,
             'Generates a config file';
-        like $out, qr/^license_key: 'foo'/m, 'Prints config with new license key';
+        like $out, qr/^license_key: 'foo'/m,
+            'Prints config with new license key';
     };
 };
 
 subtest 'run-perl' => sub {
+    # this assumes NewRelic is enabled
+    local $ENV{NEWRELIC_ENABLED}     = 1;
+    local $ENV{NEWRELIC_CONFIG_FILE} = dist_file 'NewFangle-Agent',
+        'test.yml';
+    local $ENV{NEWRELIC_ENVIRONMENT} = 'environment';
+    local $ENV{NEWRELIC_DAEMON_HOST} = '127.0.0.1:8000';
+    local $ENV{NEWRELIC_APP_NAME}    = 'unit tests';
+
     require NewFangle::Agent;
-    is test(qw( run-perl -Ilib -E ), 'say ">> $NewFangle::Agent::VERSION <<"'),
+    is test(
+        [ qw( run-perl -Ilib -E ), 'say ">> $NewFangle::Agent::VERSION <<"' ],
+        qr/failed to connect to daemon/,
```

Comment:
Won't this fail if it runs in an environment where the daemon is running?

Reply:
in a unit test environment? I guess it will in that case. what do you suggest?

Reply:
> Won't this fail if it runs in an environment where the daemon is running?

changed it to only check STDERR if we are expecting the command to fail with a non-zero exit code